### DISCUSSION TITLE
Executing a "Clean Exit" with tootctl self-destruct command taking too long. Is it normal?

### DISCUSSION THREAD

**Discussion by akibasocial** [mastodon/mastodon#30552] [Q&A]
I was looking for a way to defederate my current instance to start over on the same domain and I found that the command `tootctl self-destruct` would be exactly what I was looking for.

So I executed it two days ago, did all the steps as required:
1. Executed the `tootctl self-destruct`
2. Added the new parameter on the .env.production as requested after running the command
3. Waited

So far it's been two days with the following message showing when I re-execute the `tootctl self-destruct` command:
```
Self-destruct mode is already enabled for this Mastodon server
4 accounts are still pending deletion.
```

It seems to be the expected result of the command after correct execution, but as you can see the instance have only 4 accounts and we don't have much content, not that this would make any difference, [the documentation](https://docs.joinmastodon.org/admin/tootctl/#self-destruct) is very clear about nothing happening physically on the server installation:

> No local data is actually deleted because emptying the database or deleting the entire VPS is faster. If you run this command and then continue to operate the instance anyway, then there will be a state mismatch that might lead to glitches and issues with federation.

However my sidekick doesn't seem to have much work to do these last days.
Is it right? Should I do anything else?
How long it takes in general for the `self-destruct` concludes?

**Comment by SimplyCorbett** [comment]
So I'm going through this process myself and the actual message is that you wait until the sidekiq queue is cleared.

My suggestion would be (for anyone else doing this) to view the sidekiq count in the mastodon web interface and wait until it's completely cleared.

Unfortunately, this also means if a server is down in federation those jobs will go to the back of the queue and wait to be reprocessed. Until that server is accessible again you're not going to be finished (or, until the sidekiq jobs are marked as dead).

You can stop the process before it's done but you're not going to have a clean break. If those servers that couldn't be contacted ever come back online you will not be able to federate with them if you have already deleted the installation or the jobs have gone into the dead queue.

This is the primary reason it's recommend to just ditch the domain/subdomain and reinstall on a new domain/subdomain instead of self destructing.

**Reply by akibasocial** [reply]
I had to reboot the server, when it came back online the process triggered as it should
